GitFyle.Core.Api/Services/Foundations/Configurations/ConfigurationService.Validations.cs

```diff
@@ -0,0 +1,152 @@
+                (Rule: await IsInvalidAsync(configuration.UpdatedDate), Parameter: nameof(configuration.UpdatedDate)),
+                (Rule: await IsInvalidLengthAsync(configuration.Name, 450), Parameter: nameof(Configuration.Name)),
+
+                (Rule: await IsNotSameAsync(
+                    first: configuration.CreatedBy,
+                    second: configuration.UpdatedBy,
+                    secondName: nameof(configuration.CreatedBy)),
+
+                    Parameter: nameof(configuration.UpdatedBy)),
+
+                (Rule: await IsNotSameAsync(
+                    firstDate: configuration.CreatedDate,
+                    secondDate: configuration.UpdatedDate,
+                    secondDateName: nameof(configuration.CreatedDate)),
+
+                    Parameter: nameof(configuration.UpdatedDate)),
+
+                (Rule: await IsNotRecentAsync(configuration.CreatedDate),
+                Parameter: nameof(configuration.CreatedDate)));
+        }
+
+        private static async ValueTask<dynamic> IsInvalidLengthAsync(string text, int maxLength) => new
+        {
+            Condition = await IsExceedingLengthAsync(text, maxLength),
+            Message = $"Text exceed max length of {maxLength} characters"
+        };
+
+        private static async ValueTask<bool> IsExceedingLengthAsync(string text, int maxLength) => 
+            (text ?? string.Empty).Length > maxLength;
+
+        private async ValueTask<dynamic> IsNotRecentAsync(DateTimeOffset date)
+        {
+            var (isNotRecent, startDate, endDate) = await IsDateNotRecentAsync(date);
+
+            return new
+            {
+                Condition = isNotRecent,
+                Message = $"Date is not recent. Expected a value between {startDate} and {endDate} but found {date}"
+            };
+        }
+
+        private async ValueTask<(bool IsNotRecent, DateTimeOffset StartDate, DateTimeOffset EndDate)>
+            IsDateNotRecentAsync(DateTimeOffset date)
+        {
+            int pastSeconds = 60;
+            int futureSeconds = 0;
+
+            DateTimeOffset currentDateTime =
+                await this.dateTimeBroker.GetCurrentDateTimeOffsetAsync();
+
+            if (currentDateTime == default)
+            {
+                return (false, default, default);
+            }
+
+            TimeSpan timeDifference = currentDateTime.Subtract(date);
+            DateTimeOffset startDate = currentDateTime.AddSeconds(-pastSeconds);
+            DateTimeOffset endDate = currentDateTime.AddSeconds(futureSeconds);
+            bool isNotRecent = timeDifference.TotalSeconds is > 60 or < 0;
+
+            return (isNotRecent, startDate, endDate);
+        }
+
+        private static async ValueTask<dynamic> IsNotSameAsync(
+            DateTimeOffset firstDate,
+            DateTimeOffset secondDate,
+            string secondDateName
+            ) => new
```

Comment:
Fix the formatting to be the same as the other IsNotSameAsync

Reply:
Thank you. I have fixed it now.